### ISSUE TITLE
ratatui-minimal has strange behavior

### ISSUE THREAD

**Issue by pm100** [rhysd/tui-textarea#14]
- sometimes when started there is an extra blank line and the cursor is positioned on the second line
- if I type a single character it is repeated in the widget

![image](https://user-images.githubusercontent.com/489231/233518924-882e6ebd-643d-47fa-983d-74f3241ef613.png)

Here I simply ran it and typed hello world

It only fails on windows, works on linux (WSL unbuntu 20)

BTW - minimal (with tui) works fine

**Comment by pm100** [comment]
also checked on mac, works fine.

I tried to debug on windows (my main dev env) and it seemed that crossterm was generating extra events. This loop doesn't really involve ratatui. But maybe its something to do with how ratatui sets up crossterm, or what term.draw is doing 

```
   loop {
        term.draw(|f| {
            f.render_widget(textarea.widget(), f.size());
        })?;
        match crossterm::event::read()?.into() {
            Input { key: Key::Esc, .. } => break,
            input => {
                textarea.input(input);
            }
        }
    }
```

but I ran out of time to investigate further. I am away from my main dev env for 6 weeks, sorry I cannot add more info

**Comment by MindMayhem** [comment]
@pm100  key doubling problem occurs in vscode terminal, but not in, this is crossterm problem, as i think
but in usual cmd/powershell it doesnt doubled (idk why)

main reason - app reacts to Press and Release event both as one key event

possible solution - check for event kind before action, like

    if let Event::Key(key) = event::read()? {
                if key.kind == KeyEventKind::Press {
                        .....
                }
        }

**Comment by pm100** [comment]
I am on the road away from my windows dev box but I am pretty sure I was not in vs code terminal. I never use that for tui' development, I always have externalconsole set

________________________________

**Comment by pm100** [comment]
Will do. But in fact I am sure that I did that too. Other oddity is that it works fine with tui, it's only ratatui that is wierd.

________________________________

**Comment by pm100** [comment]
I checked. This fails if run directly from powershell command line

![minbug](https://user-images.githubusercontent.com/489231/235954146-c991fa61-a70b-4320-a61c-794a9642ab2d.gif)

and works fine with tui. 

![minbug2](https://user-images.githubusercontent.com/489231/235954984-49c12081-cc64-40d4-9747-6efe750d974e.gif)

**Comment by rhysd** [comment]
This should be fixed by 68f4e0d
